std/array.d

```diff
@@ -342,6 +342,110 @@ then the result will contain the value of the last pair for that key in r.
 See_Also: $(XREF typecons, Tuple)
  */
 
+/**
+ * Initialize a static array of the given size from the range $(D r).
+ *
+ * If `r` has more than `n` elements, only the first `n` are copied.
+ * If `r` has less than `n` elements, the remaining spaces in the static array
+ * will have default values.
+ *
+ * Params:
+ *      r = range (or aggregate with $(D opApply) function) whose elements are
+ *      copied into the static array.
+ *
+ * Returns:
+ *      static array of size `n`
+ */
+ForeachType!Range[n] array(size_t n, Range)(Range r)
```

Comment:
I've been using `T[n] arr; r.take(arr.length).copy(arr[]);` for this and I do it quite a lot, so I think this is a nice improvement.